utils/updateMetadata.ts

```diff
@@ -127,50 +119,66 @@ const processFile = async (file: string) => {
     if (difference.length > 0) {
       replacedFileContent = updateFrontMatter(
         replacedFileContent,
-        'contributors',
+        "contributors",
         JSON.stringify(contributors)
       );
     }
   }
 
   if (replacedFileContent !== fileContent) {
-    fs.writeFileSync(file, replacedFileContent, { encoding: 'utf8' });
+    fs.writeFileSync(file, replacedFileContent, { encoding: "utf8" });
     console.log(
       `Updated metatdata for: "${file}" (date=${creationDate.toISOString()},lastmod=${lastmodDate.toISOString()},contributorsDiff=[${difference.join()}])`
     );
   }
 };
 
 (async () => {
-  const files: string[] = [];
-
-  if (process.env.ALL === 'true') {
-    const allFiles = await asyncGlob(`${baseFolder}**/*.md`);
-    files.push(...allFiles);
-  } else {
-    const lastSha = process.env.GITHUB_SHA || 'HEAD';
-
-    const changelist = await childProcess
-      .execSync(`git diff-tree --no-commit-id --name-status -r ${lastSha}`)
-      .toString();
-
-    for (const line of changelist.split('\n')) {
-      const [modifier, path] = line.split(/\s+/);
-
-      // check if file is not deleted in that commit and if its an md file
-      if (path && modifier !== 'D' && path.endsWith('.md')) {
-        files.push(path);
-      }
+  new Promise<string[]>((resolve, reject) => {
+    if (process.env.ALL === "true") {
+      glob(`${baseFolder}**/*.md`, (err, matches) => {
+        if (err) {
+          reject(err);
+          return;
+        }
+
+        resolve(matches);
+      });
+    } else {
+      const lastSha = process.env.GITHUB_SHA || "HEAD";
+
+      childProcess.exec(
+        `git diff-tree --no-commit-id --name-status -r ${lastSha}`,
+        (err, stdout) => {
+          if (err) {
+            reject(err);
+            return;
+          }
+
+          const files = stdout.split("\n").reduce((arr, line) => {
+            const [modifier, path] = line.split(/\s+/);
+
+            if (path && modifier !== "D" && path.endsWith(".md")) {
+              arr.push(path);
+            }
+
+            return arr;
+          }, [] as string[]);
+
+          resolve(files);
+        }
+      );
     }
-  }
-
-  for (const file of files) {
-    if (ignoredFiles.includes(file)) continue;
-
-    await processFile(file);
-  }
-
-  console.log(`\nSuccessfully updated metadata for ${files.length} files.`);
+  })
+    .then(
+      async (files) =>
+        await Promise.all(
+          files.map(async (file) => await processFile(file))
+        ).then(() => files.length)
+    )
```

Comment:
These async and await keywords aren't necessary. And I think it would be better to keep the .then tree not nested.
```suggestion
    .then((files) => Promise.all(files.map((file) => processFile(file))))
    .then(() => files.length)
```